### PR TITLE
Add address sanitizer support for Visual Studio.

### PR DESCRIPTION
**Pull request by jpakkane**
https://devblogs.microsoft.com/cppblog/address-sanitizer-for-msvc-now-generally-available/

### CODE REVIEW

File: mesonbuild/compilers/cpp.py [hunk 1]
```diff
@@ -663,7 +663,6 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
         CPPCompiler.__init__(self, exelist, version, for_machine, is_cross,
                              info, exe_wrapper, linker=linker, full_version=full_version)
         MSVCCompiler.__init__(self, target)
-        self.base_options = {OptionKey(o) for o in ['b_pch', 'b_vscrt', 'b_ndebug']} # FIXME add lto, pgo and the like
```
Comment (dcbaker) [inline mesonbuild/compilers/cpp.py:666]: Why was this deleted, is it duplicated further down?

**Review by dcbaker — CHANGES_REQUESTED** [review]
Presumably this deserves a release snippet, and possibly something in the docs?

File: mesonbuild/compilers/cpp.py (hunk 1 above)
Comment (jpakkane) [reply mesonbuild/compilers/cpp.py:666]: No, further up. :)

It is set in the base class. I had to spend a fair bit of time head scratching why the changes I added there were not having any effect. For comparison the C version uses `base_options.update` so as not to clobber the base code.

Comment (dcbaker) [reply mesonbuild/compilers/cpp.py:666]: I'm sure I meant up, though who knows... Anyway, yes that should be using update.

**Review by dcbaker — APPROVED** [review]
LGTM thanks
